apps/remote_control/lib/lexical/remote_control/completion/struct.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Lexical.RemoteControl.Completion.Struct do
+  require Logger
+
+  def doc(full_name) when is_binary(full_name) do
+    struct = Module.concat(Elixir, full_name)
+
+    case fetch_struct_type(struct) do
+      {:ok, t} ->
+        default = default_key_values(struct)
+
+        t
+        |> Macro.to_string()
+        |> trim_parent(full_name)
+        |> maybe_put_default(default)
+        |> replace_default_symbol()
+
+      _ ->
+        "#{inspect(struct.__struct__(), pretty: true, width: 40)}"
+        |> trim_parent(full_name)
+    end
+  end
+
+  defp fetch_struct_type(struct) do
+    with {:ok, types} <- Code.Typespec.fetch_types(struct),
+         {:type, t} <- Enum.find(types, &match?({:type, {:t, _, _}}, &1)) do
```

Comment:
this relies on the typespec being called `t`. You should look for a type that is being defined in terms of a map with a`__struct__` key.